internal/apex/ctlserver.go

```diff
@@ -0,0 +1,84 @@
+package apex
+
+import (
+	"fmt"
+	"net"
+	"net/rpc"
+	"net/rpc/jsonrpc"
+	"os"
+)
+
+// TODO make this path configurable
+const UnixSocketPath = "/var/run/apex.sock"
+
+// Assumptions made in here:
+//  - ax.logger is thread-safe
+
+func (ax *Apex) CtlServerStart() error {
+	switch ax.os {
+	case Linux.String():
+		return ax.CtlServerStartLinux()
+
+	case Darwin.String():
+		ax.logger.Info("Ctl interface not yet supported on OSX")
+
+	case Windows.String():
+		ax.logger.Info("Ctl interface not yet supported on Windows")
+	}
+
+	return nil
+}
+
+type ApexCtl struct {
+	ax *Apex
+}
+
+func (ac *ApexCtl) Status(_ string, result *string) error {
+	var statusStr string
+	switch ac.ax.status {
+	case ApexStatusStarting:
+		statusStr = "Starting"
+	case ApexStatusAuth:
+		statusStr = "WaitingForAuth"
+	case ApexStatusRunning:
+		statusStr = "Running"
+	default:
+		statusStr = "Unknown"
+	}
+	res := fmt.Sprintf("Status: %s\n", statusStr)
+	if len(ac.ax.statusMsg) > 0 {
+		res += ac.ax.statusMsg
+	}
+	*result = res
+	return nil
+}
+
+func (ax *Apex) CtlServerStartLinux() error {
+	os.Remove(UnixSocketPath)
+	l, err := net.ListenUnix("unix", &net.UnixAddr{Name: UnixSocketPath, Net: "unix"})
+	if err != nil {
+		ax.logger.Error("Error creating unix socket: ", err)
+		return err
+	}
+	go func() {
+		defer l.Close()
+
+		ac := new(ApexCtl)
+		ac.ax = ax
+		err := rpc.Register(ac)
+		if err != nil {
+			ax.logger.Errorln("Error on rpc.Register(): ", err)
+		}
+
+		for {
```

Comment:
This is bothering me a little bit 😆 

If possible could we make this:
`func (ax *Apex) CtlServerRunLinux(stop <-chan struct{}) error {}`

That way:
1. The caller can pass in a channel to signal that the inner loop should stop
2. The caller will have to use `go CtlServerRunLinux()` to avoid blocking... which means that they could use some thing like `sync.WaitGroup` to ensure that async code has finished before exiting.

Reply:
done